pkg/controller/nebulascheduledbackup/nebula_scheduled_backup_control.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2023 Vesoft Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nebulascheduledbackup
+
+import (
+	"k8s.io/klog/v2"
+
+	"github.com/vesoft-inc/nebula-operator/apis/apps/v1alpha1"
+	"github.com/vesoft-inc/nebula-operator/pkg/kube"
+	"github.com/vesoft-inc/nebula-operator/pkg/util/errors"
+)
+
+type ControlInterface interface {
+	SyncNebulaScheduledBackup(bp *v1alpha1.NebulaScheduledBackup) error
+}
+
+var _ ControlInterface = (*defaultScheduledBackupControl)(nil)
+
+type defaultScheduledBackupControl struct {
+	clientSet              kube.ClientSet
+	scheduledBackupManager Manager
+}
+
+func NewBackupControl(clientSet kube.ClientSet, scheduledBackupManager Manager) ControlInterface {
+	return &defaultScheduledBackupControl{
+		clientSet:              clientSet,
+		scheduledBackupManager: scheduledBackupManager,
+	}
+}
+
+func (c *defaultScheduledBackupControl) SyncNebulaScheduledBackup(sbp *v1alpha1.NebulaScheduledBackup) error {
```

Comment:
In my view, `NebulaScheduledBackup` only need create `NebulaBackup` CR, `NebulaBackup` CR create backup job

Reply:
The status.LastBackup is used for incremental backups. In the first phase, I think it's enough to implement regular full backups, we can put this aside for now.

Reply:
Make sense, will have the scheduled backup start a go cronjob and use that to create a nebula backup when it's time by calling the create function in nebula_backup_manager.go. Will also ignore status.LastBackup for now.

Reply:
Fixed

Reply:
Several issues need to be resolved:
1. Get the next scheduled time need consider the error in edge case 
2. How to requeue after a scheduled backup complete
3. If nebulabackup objects accumulated into the hundreds, how to cleanup
4. switch phase logic is confused, it can be processed in Sync

Reply:
Several issues need to be resolved:
1. Get the next scheduled time need consider the error in edge case 
2. How to requeue after a scheduled backup complete
3. If nebulabackup objects accumulated into the hundreds, how to cleanup
4. switch phase logic is confused, it can be processed in Sync

Reply:
@MegaByte875 Got it, have a few questions on the above.
1. **Get the next scheduled time need consider the error in edge case**
    By "edge case", are you talking about invalid cron schedules (i.e. 61 * * * *)? I can add a check for this if so.
2. **How to requeue after a scheduled backup complete**
    Not sure why this is needed. The NebulaScheduledBackup reconciler automatically triggers a Nebula Backup when the cron schedule is up and waits for it to complete. Then the reconciler waits for the next time the cron scheduled is up. It'll never exit unless the NebulaScheduledBackup is paused or deleted.
3. **If nebulabackup objects accumulated into the hundreds, how to cleanup**
    Currently a nebulabackup object is automatically deleted if the backup job completed successfully. We can add parameters to the NebulaScheduledBackup object to allow the user to specify the number of completed and failed NebulaBackup jobs they'd like to keep and delete all the excess. Does this sound good to you?
4. **switch phase logic is confused, it can be processed in Sync**
   I see. Used the switch logic to be consistent with the switch statement in the NebulaBackup reconciler. Will move the ScheduledBackupPaused and ScheduledBackupFailed states into the sync function and change the switch to a if-else to clear up the confusion. We still need 2 cases there to distinguish between create and sync.

Reply:
Fixed